Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/Home.kt

```diff
@@ -108,21 +109,57 @@ enum class HomeSections(
     PROFILE(R.string.home_profile, Icons.Outlined.AccountCircle, "home/profile")
 }
 
+/**
+ * Remembers and creates an instance of [BottomBarStateHolder]
+ */
+@Composable
+fun rememberBottomBarStateHolder(navController: NavController, tabs: Array<HomeSections>) =
+    remember(navController, tabs) {
+        BottomBarStateHolder(navController, tabs)
+    }
+
+/**
+ * Responsible for holding state related to [JetsnackBottomBar] and containing UI-related logic.
+ */
+@Stable
+class BottomBarStateHolder(
+    private val navController: NavController,
+    val tabs: Array<HomeSections>
+) {
+    private val sections = HomeSections.values()
+    private val currentRoute: String?
+        get() = navController.currentDestination?.route
+
+    val routes = sections.map { it.route }
+
+    // Reading this attribute will cause recompositions when the route becomes valid or invalid
+    val isValidRoute: Boolean
+        @Composable get() =
+            navController.currentBackStackEntryAsState().value?.destination?.route in routes
+
+    val currentSection: HomeSections
+        get() = sections.first { it.route == currentRoute }
+
+    fun navigateToRoute(route: String) {
+        if (route != currentRoute) {
+            navController.navigate(route) {
+                launchSingleTop = true
+                restoreState = true
+                popUpTo(findStartDestination(navController.graph).id) {
+                    saveState = true
+                }
```

Comment:
This part doesn't really make sense to me, maybe add a comment?

Reply:
Not sure what this does, maybe @nickbutcher does

Reply:
I think that @JoseAlcerreca added this? Should we be passing the `navCotroller` around—wouldn't it be better to hoist params of whether to show bottom bar and where to navigate when selected?

Reply:
Yeah, good point. I think it's better to create a `JetsnackAppStateHolder` and remove the `JetsnackNavGraph` composable as that's what takes the `navController` as a dependency and complicates things.

Reply:
Made the changes and created the state holder for the App which makes much more sense. PTAL! Thank you

Also added more comments